sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -9,54 +10,126 @@ import com.klaviyo.coresdk.networking.requests.KlaviyoRequest
+     * @param applicationContext
+     * @param networkTimeout
+     * @param networkFlushInterval
+     * @param networkFlushDepth
+     * @param networkFlushCheckInterval
+     * @param networkUseAnalyticsBatchQueue
+     * @return
      */
-    fun setUserEmail(email: String) {
+    fun configure(
+        apiKey: String,
+        applicationContext: Context,
+        networkTimeout: Int = KlaviyoConfig.NETWORK_TIMEOUT_DEFAULT,
+        networkFlushInterval: Int = KlaviyoConfig.NETWORK_FLUSH_INTERVAL_DEFAULT,
+        networkFlushDepth: Int = KlaviyoConfig.NETWORK_FLUSH_DEPTH_DEFAULT,
+        networkFlushCheckInterval: Int = KlaviyoConfig.NETWORK_FLUSH_CHECK_INTERVAL,
+        networkUseAnalyticsBatchQueue: Boolean = KlaviyoConfig.NETWORK_USE_ANALYTICS_BATCH_QUEUE,
+    ) = apply {
+        KlaviyoConfig.Builder()
+            .apiKey(apiKey)
+            .applicationContext(applicationContext)
+            .networkTimeout(networkTimeout)
+            .networkFlushInterval(networkFlushInterval)
+            .networkFlushDepth(networkFlushDepth)
+            .networkFlushCheckInterval(networkFlushCheckInterval)
+            .networkUseAnalyticsBatchQueue(networkUseAnalyticsBatchQueue)
+            .build()
+    }
+
+    //region Fluent setters
+
+    /**
+     * Assigns an email address to the current UserInfo
+     *
+     * UserInfo is saved to keep track of current profile details and
+     * used to autocomplete analytics requests with profile identifier when not specified
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param email Email address for active user
+     */
+    fun setEmail(email: String) = apply {
         UserInfo.email = email
     }
 
     /**
-     * Creates a track request for the Klaviyo APIs and begins processing.
-     * Track requests track the triggering of a specified [KlaviyoEvent] in Klaviyo
+     * Assigns a phone number to the current UserInfo
      *
-     * @param customerProperties A map of customer property information.
-     * Defines the customer that triggered this event
-     * @param properties A map of event property information.
-     * Additional properties associated to the event that are not for identifying the customer
+     * UserInfo is saved to keep track of current profile details and
+     * used to autocomplete analytics requests with profile identifier when not specified
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param phone Phone number for active user
      */
-    fun track(
-        event: KlaviyoEvent,
-        customerProperties: KlaviyoCustomerProperties,
-        properties: KlaviyoEventProperties? = null
-    ) {
-        val request = TrackRequest(KlaviyoConfig.apiKey, event, customerProperties, properties)
-        processRequest(request)
+    fun setPhone(phone: String) = apply {
```

Comment:
Maybe this:
```suggestion
    fun setPhoneNumber(phoneNumber: String) = apply {
```
I also noticed the v3 api seems to check the format of this.

Reply:
Yes I was thinking UserInfo should have format validators on set.

Reply:
Yah might not be a bad idea however I'm not sure I want to stuff all that in the sdks. More just mentioning in case you get confused during testing.

Reply:
you're thinking better to keep the client SDK dumb, let the klaviyo api backend be responsible for validation?